### PR TITLE
feat: respect env variables

### PR DESCRIPTION
**Pull request by svipulrao98**
## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

[AI-1455]: https://altimateai.atlassian.net/browse/AI-1455?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mdesmet** [comment]
The migration is missing from the parameter to the env var. Can you try to implement that as well?

File: dbt_integration.py [hunk 1]
```diff
@@ -169,17 +169,19 @@ def wrapper(
...
+        str(project_dir)
+        if (project_dir / "profiles.yml").exists()
+        else str(Path.home() / ".dbt")
     )
 
 
 def target_dir() -> Path:
     if "DBT_TARGET_PATH" in os.environ:
-        return Path(os.environ["DBT_TARGET_PATH"]).resolve()
+        return str(Path(os.environ["DBT_TARGET_PATH"]).resolve())
```
Comment (mdesmet) [inline dbt_integration.py:184]: That won't work for windows

File: dbt_integration.py [hunk 2]
```diff
@@ -169,17 +169,19 @@ def wrapper(
...
-def default_profiles_dir(project_dir: Path) -> Path:
+def default_profiles_dir(project_dir) -> str:
     if "DBT_PROFILES_DIR" in os.environ:
-        return Path(os.environ["DBT_PROFILES_DIR"]).resolve()
+        return str(Path(os.environ["DBT_PROFILES_DIR"]).resolve())
     return (
-        project_dir if (project_dir / "profiles.yml").exists() else Path.home() / ".dbt"
+        str(project_dir)
+        if (project_dir / "profiles.yml").exists()
+        else str(Path.home() / ".dbt")
```
Comment (mdesmet) [inline dbt_integration.py:178]: That won't work for windows

Comment (svipulrao98) [reply dbt_integration.py:178]: yes, I was thinking of handling it in the python level itself, or just avoiding `str` altogether

**Pull request closed without merging**